src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,146 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private static final int STANDART_LIST_SIZE = 0;
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    public MyLinkedList() {
+        size = STANDART_LIST_SIZE;
+    }
+
+    public void isIndexExist(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Index " + index + " dosen't exist");
+        }
+    }
+
+    public Node<T> findByIndex(int index) {
+        isIndexExist(index);
+        if (head == null) {
+            return null;
+        }
+        Node<T> currentNode = head;
+        int i = 0;
+        while (i != index) {
+            currentNode = currentNode.next;
+            i++;
+        }
```

Comment:
What about adding index condition and iterate to the half of the linked list depending on the index instead of going through the whole LL?